File: Problems/406_Queue_Reconstruction_by_Height.py
from typing import List
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        people.sort()
        queue = [[-1, -1] for _ in range(len(people))]
        for man in people:
            height, infront = man
            cnt = 0
            for i, el in enumerate(queue):
                if cnt == infront and el[0]==-1:
                    queue[i] = man
                    break
                if el[0]==-1 or el[0]==height:
                    cnt += 1   
        return queue

if __name__ == '__main__':
    sol = Solution()
    people = [[7,0],[4,4],[7,1],[5,0],[6,1],[5,2]]
    print(sol.reconstructQueue(people))

File: Problems/406_Queue_Reconstruction_by_Height.cpp
class Solution {
public:
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        sort(people.begin(), people.end());
        vector<vector<int>> qu(people.size(), {-1, -1});
        for(auto man: people){
            int height = man[0];
            int infront = man[1];
            int cnt = 0;
            for(int i=0; i<qu.size(); i++){
                if(cnt==infront && qu[i][0]==-1){
                    qu[i] = man;
                    break;
                }else if(qu[i][0]==-1 || qu[i][0]==height) cnt++;

            }
        }
        return qu;
    }
};